src/OpenApi/src/Services/Schemas/OpenApiSchemaStore.cs

```diff
@@ -112,13 +121,32 @@ public void PopulateSchemaIntoReferenceCache(OpenApiSchema schema, bool captureS
                 AddOrUpdateSchemaByReference(anyOfSchema, baseTypeSchemaId);
             }
         }
-        if (schema.Properties is not null)
+
+        if (schema.Items is not null)
+        {
+            AddOrUpdateAnyOfSubSchemaByReference(schema.Items);
```

Comment:
Was this already recursive?

Reply:
It wasn't recursive before and FWIW it's not entirely recursive now.

The changes related to `AddOrUpdateAnyOfSubSchemaByReference` are specifically addressing the scenario where a polymorphic type appears as part of another type (e.g. `List<Shape>` or `Shape` as a property in a container type). The set of tests [included here](https://github.com/dotnet/aspnetcore/blob/safia/openapi-fixes/src/OpenApi/test/Microsoft.AspNetCore.OpenApi.Tests/Transformers/SchemaTransformerTests.cs#L550) cover this scenario.

Reply:
So we're going to depth-first-search traverse the schema graph until we find an `AnyOf` and then go back to the old mechanism (`AddOrUpdateSchemaByReference`)?